### PR TITLE
fix: include newVersion in composer update command

### PR DESCRIPTION
**Pull request by erikgaal**
## Changes

- Use `newVersion` in `composer update` command to make sure it only updates to the intended version, and not to a newer version.

## Context

Closes https://github.com/renovatebot/renovate/discussions/24041

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/composer/artifacts.spec.ts [hunk 1]
```diff
@@ -1183,4 +1183,26 @@ describe('modules/manager/composer/artifacts', () => {
...
+  it('includes new dependency version in update command', async () => {
+    fs.readLocalFile.mockResolvedValueOnce('{}');
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce('{}');
+    git.getRepoStatus.mockResolvedValueOnce(repoStatus);
+
+    expect(
+      await composer.updateArtifacts({
+        packageFileName: 'composer.json',
+        updatedDeps: [{ depName: 'foo', newVersion: '1.1.0' }],
```
Comment (rarkins) [inline lib/modules/manager/composer/artifacts.spec.ts:1196]: Can you also enhance the previous (existing) test to have new versions for _both_ updates and then update the matchObject for it?

Comment (erikgaal) [reply lib/modules/manager/composer/artifacts.spec.ts:1196]: Sure!

**Comment by erikgaal** [comment]
Also just ran it on my reproduction repository where it [now works fine](https://github.com/erikgaal/renovate-reproduction/pull/4)!

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/composer/artifacts.spec.ts [hunk 2]
```diff
@@ -92,14 +92,17 @@ describe('modules/manager/composer/artifacts', () => {
     expect(
       await composer.updateArtifacts({
         packageFileName: 'composer.json',
-        updatedDeps: [{ depName: 'foo' }, { depName: 'bar' }],
+        updatedDeps: [
+          { depName: 'foo', newVersion: '1.0.0' },
```
Comment (viceice) [inline lib/modules/manager/composer/artifacts.spec.ts:96]: do we still have a test, where `newVersion` isn't set? so it can test current behavior?

Comment (rarkins) [reply lib/modules/manager/composer/artifacts.spec.ts:96]: Yes, the `'disable plugins when configured locally'` one

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**